bin/dependency_services.dart

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Support for automated upgrades.
+///
+/// For now this is not a finalized interface. Don't rely on this.
+import 'dart:async';
+
+import 'package:args/args.dart';
+import 'package:args/command_runner.dart';
+import 'package:pub/src/command.dart';
+import 'package:pub/src/command/dependency_services.dart';
+import 'package:pub/src/exit_codes.dart' as exit_codes;
+import 'package:pub/src/io.dart';
+import 'package:pub/src/log.dart' as log;
+
+class DependencyServicesCommandRunner extends CommandRunner<int>
```

Comment:
Does this libary need to export anything? (besides `main`)

```suggestion
class _DependencyServicesCommandRunner extends CommandRunner<int>
```

Reply:
No - but being in bin/ I would also not expect anyone to import this...

I can make everything private if it makes us sleep better.